### PR TITLE
Add more code styling hooks and apply formatting to all files

### LINKED ISSUE #241: pydocstyle/flake8-docstring
We should integrate some checks for the style of the docstrings.

_Originally posted by @jbusecke in https://github.com/xgcm/xgcm/pull/205#discussion_r493110467_

### PR DESCRIPTION
**Pull request by andersy005**
This PR introduces new code styling pre-commit hooks. I ran the hooks on all files and this resulted in lots of changes (I hope folks are okay with this). 

 - [x] Closes #241
 - [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: xgcm/autogenerate.py [hunk 1]
```diff
@@ -1,8 +1,7 @@
-from __future__ import print_function
-from future.utils import iteritems
```
Comment (jbusecke) [inline xgcm/autogenerate.py:2]: Which hook removes the unused modules? That is awesome! Is that flake?

File: xgcm/gridops.py [hunk 2]
```diff
@@ -79,15 +79,15 @@ def _get_hfac_for_array(self, array):
...
-    ### Vertical Differences, Derivatives, and Interpolation ###
+    # Vertical Differences, Derivatives, and Interpolation
 
     def pad_zl_to_zp1(self, array, fill_value=0.0, zlname="Zl", zp1name="Zp1"):
         """Pad an array located at zl points such that it is located at
         zp1 points. An additional fill value is required for the bottom point.
 
         Parameters
         ----------
-        array : xray DataArray
```
Comment (jbusecke) [inline xgcm/gridops.py:90]: Whoa, this must be ancient! Again which of the hooks found this? Just curious.

**Review by jbusecke — APPROVED** [review]
This is awesome! Thank you so much @andersy005. Ill merge this, but was curious as to which hook was responsible for any of the changes.

File: xgcm/autogenerate.py (hunk 1 above)
Comment (andersy005) [reply xgcm/autogenerate.py:2]: `flake8` reported them, and I had to remove them manually.

**Comment by andersy005** [comment]
> but was curious as to which hook was responsible for any of the changes.

When I ran the hooks on all files, `flake8` complained the most.

**Pull request merged**